coral-service/src/main/java/com/linkedin/coral/coralservice/metastore/MetastoreProvider.java

```diff
@@ -0,0 +1,125 @@
+package com.linkedin.coral.coralservice.metastore;
+
+import com.linkedin.coral.common.HiveMetastoreClient;
+import com.linkedin.coral.common.HiveMscAdapter;
+import com.linkedin.coral.coralservice.utils.CoralUtils;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Paths;
+import java.util.Properties;
+
+import javax.naming.ConfigurationException;
+
+import com.google.common.base.Strings;
+
+import org.apache.hadoop.hive.conf.HiveConf;
+import org.apache.hadoop.hive.metastore.HiveMetaStoreClient;
+import org.apache.hadoop.hive.metastore.IMetaStoreClient;
+import org.apache.hadoop.hive.metastore.api.MetaException;
+import org.apache.hadoop.hive.ql.metadata.Hive;
+import org.apache.hadoop.hive.ql.metadata.HiveException;
+import org.apache.hadoop.hive.ql.session.SessionState;
+import org.apache.hadoop.security.UserGroupInformation;
+
+import static org.apache.hadoop.hive.conf.HiveConf.ConfVars.*;
+
+
+/**
+ * Utility class to get Hive metastore client for local hive or grid based hive using kerberos
+ */
+@SuppressWarnings("WeakerAccess")
+public class MetastoreProvider {
+
+  public static final String HIVE_METASTORE_URIS = "hive.metastore.uris";
```

Comment:
it is better to declare these constants as private, since they are not used outside this class.